src/plugins/intel_gpu/tests/test_cases/gather_gpu_test.cpp

```diff
@@ -6,53 +6,837 @@
+
+        ind=generate_random_1d<float>(b1*f1*x1*y1,-input0->get_layout().get_dim(axis),input0->get_layout().get_dim(axis)-1,1);
+        auto input1 = engine.allocate_memory({ data_types::f32, fmt, { b1, f1, x1, y1 } }); // Indexes
+
+        set_values(input0, dat);
+        set_values(input1, ind);
+
+        topology topology;
+        topology.add(input_layout("InputDictionary", input0->get_layout()));
+        topology.add(input_layout("InputText", input1->get_layout()));
+        topology.add(gather("gather", "InputDictionary", "InputText", axis, {b2,f2,y2,x2}, batch_dim, true, "", cldnn::padding(), fmt));
+
+        network network(engine, topology);
+        network.set_input_data("InputDictionary", input0);
+        network.set_input_data("InputText", input1);
+
+        auto output = network.execute().at("gather").get_memory();
+        cldnn::mem_lock<uint16_t> output_ptr(output, get_test_stream());
+        
+        auto datbfyx=dat;
+        auto indbfyx=ind;
+        for(int i=0;i<b0;i++)
+            for(int j=0;j<f0/fsv;j++)
+                for(int k=0;k<y0;k++)
+                    for(int l=0;l<x0;l++)
+                        for(int m=0;m<fsv;m++)
+                            datbfyx[i*f0*y0*x0 + (j*fsv+m)*y0*x0 + k*x0 + l]=dat[i*f0/fsv*y0*x0*fsv + j*y0*x0*fsv + k*x0*fsv + l*fsv + m];
+        for(int i=0;i<b1;i++)
+            for(int j=0;j<f1/fsv;j++)
+                for(int k=0;k<y1;k++)
+                    for(int l=0;l<x1;l++)
+                        for(int m=0;m<fsv;m++)
+                            indbfyx[i*f1*y1*x1 + (j*fsv+m)*y1*x1 + k*x1 + l]=ind[i*f1/fsv*y1*x1*fsv + j*y1*x1*fsv + k*x1*fsv + l*fsv + m];
+        
+        auto to_vec_size_t=[](const std::vector<int>& vec){return std::vector<size_t>(vec.begin(),vec.end());};
+        auto logical_dim=[](std::vector<int> a){ while(a.size()&&a.back()==1)a.pop_back(); return a.size(); };
+        ans=std::vector<FLOAT16>(b2*f2*x2*y2);
+        ngraph::runtime::reference::gather<FLOAT16,float>(
+            datbfyx.data(),
+            indbfyx.data(),
+            ans.data(),
+            ov::Shape(to_vec_size_t(input0->get_layout().get_dims())),
+            ov::Shape(to_vec_size_t(input1->get_layout().get_dims())),
+            ov::Shape(to_vec_size_t(output->get_layout().get_dims())),
+            axis,
+            batch_dim>=0?batch_dim:batch_dim+logical_dim(input1->get_layout().get_dims()));
+        
+        for(int i=0;i<b2;i++)
+            for(int j=0;j<f2/fsv;j++)
+                for(int k=0;k<y2;k++)
+                    for(int l=0;l<x2;l++)
+                        for(int m=0;m<fsv;m++)
+                            ASSERT_EQ((float)ans[i*f2*y2*x2 + (j*fsv+m)*y2*x2 + k*x2 + l],
+                            (float)float16_to_float32(output_ptr[i*f2/fsv*y2*x2*fsv + j*y2*x2*fsv + k*x2*fsv + l*fsv + m]));
+    }
+    // void TearDown() override {}
+};
+TEST_F(gather8fsv4ym1Fixt, a){}
+TEST_F(gather8fsv4ym1Fixt, b){}
+TEST_F(gather8fsv4ym1Fixt, c){}
```

Comment:
What is these a,b,c?

Reply:
Updated to parameterized version & fixed.